sklearn/linear_model/tests/test_least_angle.py

```diff
@@ -938,3 +940,42 @@ def test_lars_numeric_consistency(LARS, has_coef_path, args):
     if has_coef_path:
         assert_allclose(model_64.coef_path_, model_32.coef_path_, rtol=rtol, atol=atol)
     assert_allclose(model_64.intercept_, model_32.intercept_, rtol=rtol, atol=atol)
+
+
+@pytest.mark.parametrize("criterion", ["aic", "bic"])
+def test_lassolarsic_alpha_selection(criterion):
+    """Check that we properly compute the AIC and BIC score.
+
+    In this test, we reproduce the example of the Fig. 2 of Zou et al.
+    In this example, only 7 features should be selected.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/14566
+    https://github.com/scikit-learn/scikit-learn/issues/17145
```

Comment:
```suggestion
```
I think this is not needed here. This PR closes those issues and the test has a clear description.